src/utils/operator-job.ts

```diff
@@ -147,22 +158,62 @@ export abstract class OperatorJobAwareCommand extends HealthCheck {
          which index position inside of the pod doe they hold (used for fallback operator calculations),
          the current pod size (used for fallback operator calculations).
   */
-  async updateOperatorStatus(network: string): Promise<void> {
+  async updateOperatorStatus(network: string): Promise<boolean> {
     const contract: Contract = this.networkMonitor.operatorContract.connect(this.networkMonitor.providers[network])
-    this.operatorStatus.active[network] = !BigNumber.from(
-      await contract.getBondedAmount(this.operatorStatus.address),
-    ).isZero()
-    this.operatorStatus.currentPod[network] = BigNumber.from(
-      await contract.getBondedPod(this.operatorStatus.address),
-    ).toNumber()
-    this.operatorStatus.podIndex[network] = BigNumber.from(
-      await contract.getBondedPodIndex(this.operatorStatus.address),
-    ).toNumber()
+
+    // A flag indicating the success of all contract calls
+    let allCallsSuccessful = true
+
+    const contractCall = async <T>(method: () => Promise<T>, errorMessage: string): Promise<T | null> => {
+      try {
+        return await method()
+      } catch (error: any) {
+        this.networkMonitor.structuredLogError(network, errorMessage + ': ' + error)
+        allCallsSuccessful = false // Mark the flag as false on any error
+        return null
+      }
+    }
+
+    const bondedAmount = await contractCall(
+      () => contract.getBondedAmount(this.operatorStatus.address),
+      'Error getting Bonded Amount',
+    )
+
+    if (bondedAmount) {
+      this.operatorStatus.active[network] = !BigNumber.from(bondedAmount).isZero()
+    }
+
+    const bondedPod = await contractCall(
+      () => contract.getBondedPod(this.operatorStatus.address),
+      'Error getting Bonded Pod',
+    )
+
+    if (bondedPod) {
+      this.operatorStatus.currentPod[network] = BigNumber.from(bondedPod).toNumber()
+    }
+
+    const bondedPodIndex = await contractCall(
+      () => contract.getBondedPodIndex(this.operatorStatus.address),
+      'Error getting Bonded Pod Index',
+    )
+
+    if (bondedPodIndex) {
+      this.operatorStatus.podIndex[network] = BigNumber.from(bondedPodIndex).toNumber()
+    }
+
     if (this.operatorStatus.currentPod[network] > 0) {
-      this.operatorStatus.podSize[network] = BigNumber.from(
-        await contract.getPodOperatorsLength(this.operatorStatus.currentPod[network]),
-      ).toNumber()
+      const podOperatorsLength = await contractCall(
+        () => contract.getPodOperatorsLength(this.operatorStatus.currentPod[network]),
+        'Error getting Pod Operators Length',
+      )
+
+      if (podOperatorsLength) {
+        this.operatorStatus.podSize[network] = BigNumber.from(podOperatorsLength).toNumber()
+      }
     }
+
+    // Return the final status of all contract calls
+    return allCallsSuccessful
   }
 
   async checkJobStatus(operatorJobHash: string, tags?: (string | number)[]): Promise<void> {
```

Comment:
@alexanderattar We may want to catch the else case in this function. 

Also, i noticed that the log on line 223 is very rarely called. In datadog there are only 13 instances in the last month across all networks. 

We may also want to log the size of `this.operatorJobs` here as well just in case.